packages/ra-ui-materialui/src/field/FileField.tsx

```diff
@@ -37,7 +41,7 @@ export const FileField = (props: FileFieldProps) => {
         ...rest
     } = props;
     const record = useRecordContext(props);
-    const sourceValue = get(record, source);
+    const sourceValue = get(record, source) as string;
```

Comment:
Why don't you use `toString()` as in `RichTextField` and `TextField`? Seems a better way to make sure that what you get is a string. 

Same for other `as string` forced casting.